### PR TITLE
added src boolean to NetmaskGroupRule to match destination address

### PR DESCRIPTION
**Pull request by skoef**
In order to solve #4104, I've created this possible solution to the problem. Also split `NetmaskGroupRule` into `NMGRule` and its subclass `NetmaskGroupRule` as @cmouse suggested.

### CODE REVIEW

File: pdns/dnsdist-lua.cc [hunk 1]
```diff
@@ -779,8 +779,8 @@ vector<std::function<void(void)>> setupLua(bool client, const std::string& confi
       return std::shared_ptr<DNSRule>(new SuffixMatchNodeRule(smn, quiet ? *quiet : false));
     });
 
-  g_lua.writeFunction("NetmaskGroupRule", [](const NetmaskGroup& nmg) {
-      return std::shared_ptr<DNSRule>(new NetmaskGroupRule(nmg));
+  g_lua.writeFunction("NetmaskGroupRule", [](const NetmaskGroup& nmg, bool src = true) {
```
Comment (rgacogne) [inline pdns/dnsdist-lua.cc]: With the Lua wrapper we use, you need to make `src` a `boost::optional<bool>` if you want it to be optional, which is a good idea to keep compatibility. You'll then have to check whether it is set before using it. There is an example just above in `SuffixMatchNodeRule`.

**Comment by rgacogne** [comment]
I'm not sure what the use of the `NMGRule` class with the current design. Perhaps two separate classes `NetmaskGroupRule` and `DestinationNetmaskGroupRule` both inheriting from `NMGRule` would make more sense?

**Comment by skoef** [comment]
I believe that was my initial suggestion on IRC, but @cmouse made this suggestion which I went for. What to do now?

**Comment by rgacogne** [comment]
Alright, it does not matter much, let's merge that. Thank you!

**Pull request merged**